Taskfile.yml

```
@@ -206,59 +212,99 @@ tasks:
     generates: ["{{.CHECKSUM_FILE}}"]
 
   webui-nodejs:
+    internal: true
     vars:
       CHECKSUM_FILE: "{{.G_BUILD_DIR}}/{{.TASK}}.md5"
       OUTPUT_DIR: "{{.G_WEBUI_NODEJS_BUILD_DIR}}"
+    cmds:
+      - task: "nodejs"
+        vars:
+          CHECKSUM_FILE: "{{.CHECKSUM_FILE}}"
+          NODEJS_VERSION: "v14.21.3"
+          OUTPUT_DIR: "{{.OUTPUT_DIR}}"
+
+  core-submodules:
+    internal: true
+    dir: "components/core"
+    vars:
+      CHECKSUM_FILE: "{{.G_BUILD_DIR}}/{{.TASK}}.md5"
+      OUTPUT_DIR: "submodules"
     deps:
       - "init"
       - task: "validate-checksum"
         vars:
           CHECKSUM_FILE: "{{.CHECKSUM_FILE}}"
           DATA_DIR: "{{.OUTPUT_DIR}}"
     cmds:
-      - task: "nodejs"
-        vars:
-          NODEJS_VERSION: "v14.21.3"
-          OUTPUT_DIR: "{{.OUTPUT_DIR}}"
+      - "tools/scripts/deps-download/download-all.sh"
       # This command must be last
       - task: "compute-checksum"
         vars:
           DATA_DIR: "{{.OUTPUT_DIR}}"
           OUTPUT_FILE: "{{.CHECKSUM_FILE}}"
     sources:
       - "{{.TASKFILE}}"
+      - ".gitmodules"
+      - "tools/scripts/deps-download/**/*"
     generates: ["{{.CHECKSUM_FILE}}"]
 
-  core-submodules:
+  download-and-extract-tar:
     internal: true
-    dir: "components/core"
+    requires:
+      vars: ["CHECKSUM_FILE", "EXTRACTED_DIR_NAME", "TAR_NAME", "OUTPUT_DIR", "URL_PREFIX"]
+    label: "{{.TASK}}-{{.TAR_NAME}}"
     vars:
-      CHECKSUM_FILE: "{{.G_BUILD_DIR}}/{{.TASK}}.md5"
-      OUTPUT_DIR: "submodules"
+      OUTPUT_TMP_DIR: "{{.OUTPUT_DIR}}-tmp"
+      EXTRACTED_DIR: "{{.OUTPUT_TMP_DIR}}/{{.EXTRACTED_DIR_NAME}}"
+      TAR_PATH: "{{.OUTPUT_TMP_DIR}}/{{.TAR_NAME}}"
     deps:
       - "init"
       - task: "validate-checksum"
         vars:
           CHECKSUM_FILE: "{{.CHECKSUM_FILE}}"
           DATA_DIR: "{{.OUTPUT_DIR}}"
     cmds:
-      - "tools/scripts/deps-download/download-all.sh"
+      - "rm -rf '{{.OUTPUT_DIR}}' '{{.OUTPUT_TMP_DIR}}'"
+      - "mkdir -p '{{.OUTPUT_TMP_DIR}}'"
+      - >-
+        curl --fail --location --show-error
+        "{{.URL_PREFIX}}/{{.TAR_NAME}}"
+        --output "{{.TAR_PATH}}"
+      - "tar xf '{{.TAR_PATH}}' --directory '{{.OUTPUT_TMP_DIR}}'"
+      - "mv '{{.EXTRACTED_DIR}}' '{{.OUTPUT_DIR}}'"
+      - "rm -rf '{{.OUTPUT_TMP_DIR}}'"
       # This command must be last
       - task: "compute-checksum"
         vars:
           DATA_DIR: "{{.OUTPUT_DIR}}"
           OUTPUT_FILE: "{{.CHECKSUM_FILE}}"
-    sources:
-      - "{{.TASKFILE}}"
-      - ".gitmodules"
-      - "tools/scripts/deps-download/**/*"
+    sources: ["{{.TASKFILE}}"]
     generates: ["{{.CHECKSUM_FILE}}"]
 
+  meteor:
+    run: "once"
+    preconditions:
+      # NOTE: Meteor 2.x does not support AArch64 on Linux
+      - "test '$(uname -m)' != 'aarch64' || (test '$(uname -s)' != 'Linux')"
```

```suggestion
      - sh: >-
          (test "$(uname -m)" != "aarch64") || (test "$(uname -s)" != "Linux")
        msg: "Meteor 2.x does not support aarch64 on Linux"
```

(Need to switch to double-quotes otherwise the variable expansion will be interpreted literally.)